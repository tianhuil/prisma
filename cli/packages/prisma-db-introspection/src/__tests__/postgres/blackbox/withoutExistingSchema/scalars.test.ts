import testSchema from '../common'

describe('Introspector', () => {
  test('text columns', async () => {
    await testSchema(`CREATE TABLE "Strings" (
      "pk" varchar(55) NOT NULL PRIMARY KEY,
      "a" char(1) DEFAULT NULL UNIQUE,
      "aa" char(1) NOT NULL UNIQUE,
      "b" varchar(255) DEFAULT NULL,
      "c" text DEFAULT 'abc',
      "cc" text DEFAULT 'abc' NOT NULL,
      "d" char(1) NOT NULL,
      "e" varchar(255) NOT NULL,
      "f" text NOT NULL,
      "g" uuid DEFAULT NULL,
      "h" uuid NOT NULL,
      constraint foo unique(b)
      );`)
  })

  test('int columns', async () => {
    await testSchema(`CREATE TABLE "Ints" (
      "a" smallint DEFAULT 3,
      "b" integer DEFAULT NULL,
      "c" bigint DEFAULT NULL,
      "d" smallint NOT NULL,
      "e" integer NOT NULL,
      "f" bigint NOT NULL
      );`)
  })

  test('float columns', async () => {
    await testSchema(`CREATE TABLE "Floats" (
      "a" real DEFAULT 3.4,
      "b" double precision DEFAULT NULL,
      "c" float4 DEFAULT NULL,
      "d" float8 DEFAULT NULL,
      "e" real NOT NULL,
      "f" double precision NOT NULL,
      "g" float4 NOT NULL,
      "h" float8 NOT NULL,
      "i" numeric DEFAULT NULL,
      "j" numeric NOT NULL
      );`)
  })

  test('boolean columns', async () => {
    await testSchema(`CREATE TABLE "Booleans" (
      "a" boolean DEFAULT NULL,
      "b" boolean NOT NULL
      );`)
  })

  test('DateTime columns', async () => {
    await testSchema(`CREATE TABLE "Booleans" (
      "a" timestamp DEFAULT NULL,
      "b" timestamp NOT NULL,
      "c" timestamp NOT NULL DEFAULT now()
      );`)
  })

  test('Json columns', async () => {
    await testSchema(`CREATE TABLE "Booleans" (
      "a" json DEFAULT NULL,
      "b" json NOT NULL
      );`)
  })

  test('Unmapped columns', async () => {
    await testSchema(`CREATE TABLE "Unmapped" (
      "a" jsonb,
      "c" tsvector,
      "d" date[]
      );`)
  })

  test('uuid columns', async () => {
    await testSchema(`CREATE TABLE "UUIDs" (
      "pk" uuid NOT NULL PRIMARY KEY,
      "nk" uuid NOT NULL,
      "tk" uuid
      );`)
  })
})
