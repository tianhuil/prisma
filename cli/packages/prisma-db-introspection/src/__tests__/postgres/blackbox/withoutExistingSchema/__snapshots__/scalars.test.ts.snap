// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introspector DateTime columns 1`] = `
"type Booleans {
  a: DateTime
  b: DateTime!
  c: DateTime!
}"
`;

exports[`Introspector DateTime columns 2`] = `
"type Booleans {
  a: DateTime
  b: DateTime!
  c: DateTime!
}"
`;

exports[`Introspector Json columns 1`] = `
"type Booleans {
  a: Json
  b: Json!
}"
`;

exports[`Introspector Json columns 2`] = `
"type Booleans {
  a: Json
  b: Json!
}"
`;

exports[`Introspector Unmapped columns 1`] = `
"type Unmapped {
  a: Json
  # Type tsvector is not supported
  # c: tsvector
  d: DateTime
}"
`;

exports[`Introspector Unmapped columns 2`] = `
"type Unmapped {
  a: Json
  # Type tsvector is not supported
  # c: tsvector
  d: DateTime
}"
`;

exports[`Introspector boolean columns 1`] = `
"type Booleans {
  a: Boolean
  b: Boolean!
}"
`;

exports[`Introspector boolean columns 2`] = `
"type Booleans {
  a: Boolean
  b: Boolean!
}"
`;

exports[`Introspector float columns 1`] = `
"type Floats {
  a: Float @default(value: 3.4)
  b: Float
  c: Float
  d: Float
  e: Float!
  f: Float!
  g: Float!
  h: Float!
  i: Float
  j: Float!
}"
`;

exports[`Introspector float columns 2`] = `
"type Floats {
  a: Float @default(value: 3.4)
  b: Float
  c: Float
  d: Float
  e: Float!
  f: Float!
  g: Float!
  h: Float!
  i: Float
  j: Float!
}"
`;

exports[`Introspector int columns 1`] = `
"type Ints {
  a: Int @default(value: 3)
  b: Int
  c: Int
  d: Int!
  e: Int!
  f: Int!
}"
`;

exports[`Introspector int columns 2`] = `
"type Ints {
  a: Int @default(value: 3)
  b: Int
  c: Int
  d: Int!
  e: Int!
  f: Int!
}"
`;

exports[`Introspector text columns 1`] = `
"type Strings {
  id: ID! @pgColumn(name: \\"pk\\") @unique
  a: String @unique
  aa: String! @unique
  b: String @unique
  c: String @default(value: \\"abc\\")
  cc: String! @default(value: \\"abc\\")
  d: String!
  e: String!
  f: String!
  g: UUID
  h: UUID!
}"
`;

exports[`Introspector text columns 2`] = `
"type Strings {
  pk: ID! @id
  a: String @unique
  aa: String! @unique
  b: String @unique
  c: String @default(value: \\"abc\\")
  cc: String! @default(value: \\"abc\\")
  d: String!
  e: String!
  f: String!
  g: UUID
  h: UUID!
}"
`;

exports[`Introspector uuid columns 1`] = `
"type UUIDs {
  id: UUID! @pgColumn(name: \\"pk\\") @unique
  nk: UUID!
  tk: UUID
}"
`;

exports[`Introspector uuid columns 2`] = `
"type UUIDs {
  pk: UUID! @id
  nk: UUID!
  tk: UUID
}"
`;
